src/ResourceManager/Automation/Commands.Automation/Common/Constants.cs

```diff
@@ -59,7 +59,7 @@ public class AutomationAccountState
 
         public const string PsCommandParamDepth = "Depth";
 
-        public const int PsCommandValueDepth = 10;
```

Comment:
Since default it 2 do we need this property?

Reply:
Do we have a test that covers the issue that got fixed here?

Reply:
@safeermohammed  We will set the depth to 4

Reply:
@ikanni What is the deciding factor on the depth being changed first from 10 to 2 and then 2 to 4 in the second iteration?

Reply:
In ConvertTo-Json the default depth is 2, so we thought the depth 2 is good enough. We realized that we may not  support an complex objects if we set to depth to 2. So changed the depth to 4.

Reply:
@ikanni Yes, but it is currently set to 10, and that doesn't seem to be enough depth, and I don't see any additional references to this value (did I miss them?), so how is changing this static value going to change the serialization depth?

Reply:
We have changed from 10 to 4. PsCommandValueDepth  is used in  PowerShellJsonConverter. Serialize. Please check https://github.com/Azure/azure-powershell/blob/b6203d65b3b12567ccef0003f954206b494c1f69/src/ResourceManager/Automation/Commands.Automation/Common/PowershellJsonConverter.cs#L36

Reply:
@ikanni Please see your previous comment.  You are saying that depth is changing from 2 to 4, but in fact, it is changing from 10 to 4.  How is that addressing the underlying issue?  If, as the customer is reporting, the depth is too shallow, how is decreasing the depth addressing their problem?

Reply:
@ikanni Yes, I was confiused by your previous comment that you were changing it from 10 to 4.  Now that it is clear that depoth 10 is causing a timeout, reducing the depth makes sense as a solution.